src/Lykke.Service.Assets.Services/AssetGroupService.cs

```diff
@@ -193,18 +190,21 @@ public async Task<bool> SwiftDepositEnabledAsync(string clientId, bool isIosDevi
 
         public async Task UpdateGroupAsync(IAssetGroup group)
         {
-            foreach (var assetId in await GetAssetIdsForGroupAsync(group.Name))
-            {
-                await _assetGroupAssetLinkRepository.UpdateAsync(assetId, group);
-            }
-
-            foreach (var clientId in await GetClientIdsForGroupAsync(group.Name))
-            {
-                await _clientAssetGroupLinkRepository.UpdateAsync(clientId, group);
-                await _assetGroupClientLinkRepository.UpdateAsync(clientId, group);
-            }
-
-            await _assetGroupRepository.UpdateAsync(group);
+            var tasks = new List<Task>();
+            var assetLinksTasks = (await GetAssetIdsForGroupAsync(group.Name)).Select(assetId =>
+                _assetGroupAssetLinkRepository.UpdateAsync(assetId, group));
+            tasks.AddRange(assetLinksTasks);
+
+            var clientIds = (await GetClientIdsForGroupAsync(group.Name)).ToArray();
+            var clientAssetGroupTasks =
+                clientIds.Select(clientId => _clientAssetGroupLinkRepository.UpdateAsync(clientId, group));
+            tasks.AddRange(clientAssetGroupTasks);
+            var assetGroupTasks =
+                clientIds.Select(clientId => _assetGroupClientLinkRepository.UpdateAsync(clientId, group));
+            tasks.AddRange(assetGroupTasks);
+
+            tasks.Add(_assetGroupRepository.UpdateAsync(group));
+            await Task.WhenAll(tasks.ToArray());
```

Comment:
Does the order of task execution matter in this place? I mean, before we had the call to
`await _assetGroupClientLinkRepository.UpdateAsync(clientId, group);`
just after the call to
`await _clientAssetGroupLinkRepository.UpdateAsync(clientId, group);`
with the same `(clientId, group)`.
And also we had
`await _assetGroupAssetLinkRepository.UpdateAsync(assetId, group);`
at the very beginning, and the
`await _assetGroupRepository.UpdateAsync(group);`
at the end. Now the order of the execution will be unknown and may differ from time to time.

Reply:
The order doesn't matter here. The only problem we can have here - the lack of transactionality. But I don't want to fix it right now. My goal is to reduce requests latency.